docs/configuration-parameters.md

```diff
@@ -149,6 +149,7 @@ This section provides the details of all the configurations required for the Car
 | carbon.max.pagination.lru.cache.size.in.mb | -1 | Maximum memory **(in MB)** upto which the SDK pagination reader can cache the blocklet rows. Suggest to configure as multiple of blocklet size. Default value of -1 means there is no memory limit for caching. Only integer values greater than 0 are accepted. |
 | carbon.partition.max.driver.lru.cache.size | -1 | Maximum memory **(in MB)** upto which driver can cache partition metadata. Beyond this, least recently used data will be removed from cache before loading new set of values.
 | carbon.mapOrderPushDown.<db_name>_<table_name>.column| empty | If order by column is in sort column, specify that sort column here to avoid ordering at map task . |
+| carbon.metacache.expiration.seconds | Long.MAX_VALUE | Expiration time **(in seconds)** for tableInfo cache in CarbonMetadata and tableModifiedTime in CarbonFileMetastore, after the time configured since last access to the cache entry, tableInfo and tableModifiedTime will be removed from each cache. Recent access will refresh the timer. Default value of Long.MAX_VALUE means the cache will not be expired by time. **NOTE:** At the time when cache is being expired, queries on the table may fail with NullPointerException. |
```

Comment:
During your test did you face NullPointerException as mentioned in this doc?

Reply:
Yes, if we set carbon.metacache.expiration.seconds to very small like 1s, UT will fail with NullPointerException.

Reply:
then its a problem, because queries would fail randomly.

@akashrn5 @ajantha-bhat what do you suggest?

Reply:
In one application itself if we do drop table and query concurrently what is the behavior ? 
a) Giving NPE is not good idea. If the other application has OOM, user can call drop metacache ?
b) Also for multiple application scenario, user should use index server for caching, which will avoid these problems as only one cache will be present for all the application.

Reply:
agree with @kunal642 , if randomly it fails then it will give false info to user and everytime checking whether its because of this time expiration or some other genuine NullPointer errors

Reply:
I think this feature is just for those customers who care about the memory leak. So if the customer doesn't want memory leak, he can choose a proper expired time which at the time of cache expired, the table will never be queried. For example, customer will have a new table every day, and this table will be queried only at that day. So he can choose 1 week as the property value so that this table cache will only leak for one week and after one week will be cleared.
@ajantha-bhat the memory leak is not about the index cache, it is table metadata cache.

Reply:
> c) or may be we should enhance drop table to clean these entries if exists when called from application 2 even though table doesn't exist but stale metadata exist in memory.

there is one method called `removeStaleTimeStampEntries` which removes invalid timestamp entries in memory, but this will be called in drop table command, and in other application until and unless user calls any drop table (basically any valid drop table), else it wont be cleared

Reply:
@kunal642 https://issues.apache.org/jira/browse/CARBONDATA-4098 raised one jira to tack the null pointer issue.

Reply:
ok, ill merge this as the default behavior is not to remove cache.

Reply:
please fix the NPE in a separate PR